### PR TITLE
Code Cleanup - Github source

### PR DESCRIPTION
**Pull request by dat-a-man**
All the rest of the API helper functions were defined two times, one in "__init__.py" and one in "helper.py" as well. This was the issue mentioned in issue #294. Now, all the helper functions are moved to "helper.py".

# Tell us what you do here

- [ ] implementing verified source (please link a relevant issue labelled as `verified source`)
- [ ] fixing a bug (please link a relevant bug report)
- [X] improving, documenting, customizing existing source (please link an issue or describe below)
- https://github.com/dlt-hub/verified-sources/issues/294
- [ ] anything else (please link an issue or describe below)

# Relevant issue

issue # 294

# More PR info
- The rest API helper functions were deleted from "__init__.py"  and updated in "helpers.py". 
- Earlier functions in "__init__.py" were used for the source, now, the same functions are called from "helpers.py".
- The functions in "__init__.py" were used in the source, so those were given preference over the functions in "helpers.py"
- Most of the functions were the same except "_run_graphql_query, _get_reactions_data"

### CODE REVIEW

File: sources/github/helpers.py [hunk 1]
```diff
@@ -22,10 +22,15 @@ def _get_auth_header(access_token: str) -> StrAny:
 #
 # Rest API helpers
 #
-def get_rest_pages(access_token: str, query: str) -> Iterator[List[StrAny]]:
-    url = REST_API_BASE_URL + query
-    while True:
+def _get_rest_pages(access_token: str, query: str) -> Iterator[List[StrAny]]:
```
Comment (AstrakhantsevaAA) [inline sources/github/helpers.py:25]: you import this function to another script, so imo prefix "_", which means "private function", doesn't make sense. 
you can keep it like `get_rest_pages`.

same for `get_reactions_data`

**Review by AstrakhantsevaAA — APPROVED** [review]

**Review by AstrakhantsevaAA — CHANGES_REQUESTED** [review]
check unused import, also run `black` command for auto-formatting

File: sources/github/helpers.py (hunk 1 above)
Comment (dat-a-man) [reply sources/github/helpers.py:25]: okay

Comment (dat-a-man) [reply sources/github/helpers.py:25]: done.

File: sources/github/helpers.py [hunk 2]
```diff
@@ -104,11 +110,15 @@ def _extract_nested_nodes(item: DictStrAny) -> DictStrAny:
...
     access_token: str, query: str, variables: DictStrAny
 ) -> Tuple[StrAny, StrAny]:
-    data = requests.post(
-        GRAPHQL_API_BASE_URL,
-        json={"query": query, "variables": variables},
-        headers=_get_auth_header(access_token),
-    ).json()
+    def _request() -> requests.Response:
+        r = requests.post(
+            "https://api.github.com/graphql",
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply sources/github/helpers.py:115]: done.

File: sources/github/helpers.py [hunk 3]
```diff
@@ -22,10 +22,15 @@ def _get_auth_header(access_token: str) -> StrAny:
...
-def get_rest_pages(access_token: str, query: str) -> Iterator[List[StrAny]]:
-    url = REST_API_BASE_URL + query
-    while True:
+def _get_rest_pages(access_token: str, query: str) -> Iterator[List[StrAny]]:
+    def _request(url: str) -> requests.Response:
         r = requests.get(url, headers=_get_auth_header(access_token))
+        print(f"got page {url}, requests left: " + r.headers["x-ratelimit-remaining"])
+        return r
+
+    url = "https://api.github.com" + query
```
[the remark this answers is not in the document]
Comment (dat-a-man) [reply sources/github/helpers.py:31]: done.

File: sources/github/helpers.py [hunk 4]
```diff
@@ -23,16 +23,23 @@ def _get_auth_header(access_token: str) -> StrAny:
...
 def get_rest_pages(access_token: str, query: str) -> Iterator[List[StrAny]]:
-    url = REST_API_BASE_URL + query
+    def _request(page_url: str) -> requests.Response:
+        r = requests.get(page_url, headers=_get_auth_header(access_token))
+        print(
+            f"got page {page_url}, requests left: " + r.headers["x-ratelimit-remaining"]
+        )
+        return r
+
+    next_page_url = "https://api.github.com" + query
```
Comment (AstrakhantsevaAA) [inline sources/github/helpers.py:33]: use REST_API_BASE_URL instead of "https://api.github.com"

Comment (dat-a-man) [reply sources/github/helpers.py]: done.

**Review by AstrakhantsevaAA — APPROVED** [review]

**Pull request merged**